aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -230,12 +252,17 @@ ShapeAndDims canonicalize_fft_shape_and_dim_args(
 
 // Complex to complex n-dimensional fft
 Tensor fftn_c2c(
-    const Tensor& input, IntArrayRef shape, IntArrayRef dim,
-    c10::optional<std::string> norm_str, bool forward) {
+    Tensor out, const Tensor& input, IntArrayRef shape,
+    IntArrayRef dim, c10::optional<std::string> norm_str, bool forward) {
   TORCH_CHECK(input.is_complex(), "Expected a complex input tensor to FFT");
   Tensor x = resize_fft_input(input, dim, shape);
   const auto norm = norm_from_string(norm_str, forward);
-  return at::_fft_c2c(x, dim, static_cast<int64_t>(norm), forward);
+  if (out.defined()) {
+    TORCH_CHECK(out.is_complex(), "fft expects a complex output tensor, but got ", out.scalar_type());
```

Comment:
"fft expects" -> "fftn expects"?

Reply:
Actually, see comment above about passing a string to specialize the error message appropriately.